samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -915,22 +929,26 @@ public TaskRestoreCallable(SamzaContainerMetrics samzaContainerMetrics, TaskName
     @Override
     public Void call() {
       long startTime = System.currentTimeMillis();
-      LOG.info("Starting stores in task instance {}", this.taskName.getTaskName());
-      taskRestoreManager.restore();
-
-      // Stop all persistent stores after restoring. Certain persistent stores opened in BulkLoad mode are compacted
-      // on stop, so paralleling stop() also parallelizes their compaction (a time-intensive operation).
-      taskRestoreManager.stopPersistentStores();
-
-      long timeToRestore = System.currentTimeMillis() - startTime;
-
-      if (this.samzaContainerMetrics != null) {
-        Gauge taskGauge = this.samzaContainerMetrics.taskStoreRestorationMetrics().getOrDefault(this.taskName, null);
-
-        if (taskGauge != null) {
-          taskGauge.set(timeToRestore);
+      try {
+        LOG.info("Starting stores in task instance {}", this.taskName.getTaskName());
+        taskRestoreManager.restore();
+      } catch (InterruptedException e) {
+        LOG.warn("Received an interrupt during store restoration for task: {}.", this.taskName.getTaskName());
```

Comment:
Since the exception thrown within will be swallowed by the executor service, I didn't propagate it upstream. Right now, the restore managers are interrupted externally by `ContainerStorageManager` which only happens in the event of the container thread getting interrupted. In the case outlined, I already bubble up the exception upstream, although I agree its still better to propagate this up.

However, any exceptions that happen within `TaskRestoreManager` are not handled today. In order to do so, we need to have uncaught exception handler for the executor service and somehow notify container thread. Let me create a follow up ticket and add a todo.

Reply:
I didn't realize it was within the executor service. So would an exception be propagated up through to `future.get()` through an `ExecutionException`? Is it possible that the actual execution gets interrupted first, causing the `future.get()` to throw an `ExecutionException`, before `future.get()` itself directly throws the `InterruptedException`?

Reply:
My bad. It will throw all checked exception wrapped with `ExceutionException`.
However, it will be the container thread that gets interrupted first during `future.get()` and only as a result of the interruption, we end up interrupting the restore thread.

Reply:
Ok, seems like it might be ok here to just log the `InterruptedException` and not propagate it.
Might be good to put a comment here to clarify what is expected to happen.